app/src/test/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailScreenTest.kt

```diff
@@ -0,0 +1,178 @@
+            onNodeWithTag(testTag = TestTagDetailLogo).assertIsDisplayed()
+            onNodeWithTag(testTag = TestTagDetailCurrentPrice).assertTextEquals(
+                "$${coinDetailUiModel.currentPrice.toFormattedString()}"
+            )
+            onNodeWithTag(testTag = TestTagDetailPriceChangePercent).assertTextEquals(
+                "${abs(coinDetailUiModel.priceChangePercentage24hInCurrency).toFormattedString()}%"
+            )
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the Chart properly`() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagDetailLineChart).assertExists()
+            onNodeWithTag(testTag = TestTagDetailChartInterval).assertExists()
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the CoinInfo properly`() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagDetailCoinInfo).assertExists()
+
+            with(onAllNodesWithTag(testTag = TestTagDetailItemTitle)) {
+                onFirst().assertTextEquals(marketCapTitle)
+                this[1].assertTextEquals(allTimeHighTitle)
+                onLast().assertTextEquals(allTimeLowTitle)
+            }
+
+            with(onAllNodesWithTag(testTag = TestTagDetailItemPrice)) {
+                onFirst().assertTextEquals("$${coinDetailUiModel.marketCap.toFormattedString()}")
+                this[1].assertTextEquals("$${coinDetailUiModel.ath.toFormattedString()}")
+                onLast().assertTextEquals("$${coinDetailUiModel.atl.toFormattedString()}")
+            }
+
+            with(onAllNodesWithTag(
+                testTag = TestTagDetailItemPriceChange
+            )) {
+                onFirst().onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.marketCapChangePercentage24h).toFormattedString()}%"
+                )
+                this[1].onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.athChangePercentage).toFormattedString()}%"
+                )
+                onLast().onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.atlChangePercentage).toFormattedString()}%"
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the SellBuyGroup properly`() {
+        initViewModel()
+
+        composeAndroidTestRule.onNodeWithTag(testTag = TestTagDetailSellBuyGroup).assertExists()
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail failed, it shows the Toast properly`() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flow {
+            throw Throwable(errorGeneric)
+        }
+
+        initViewModel()
+
```

Comment:
```suggestion
        }
```

Reply:
In this case, I don't think we can. We need to initViewModel after mocking a fail case for `mockGetCoinDetailUseCase`.

Reply:
It works all cases in `DetailScreen` for the Unit Test @luongvo I will update cc @kaungkhantsoe 👍

Reply:
@hoangnguyen92dn The test doesn't fail, You can check another test case 🙏  If I'm not wrong, it warning this all case

Reply:
@Wadeewee I found the cause finally, there is a lack of mocking answers for the `mockGetCoinPricesUseCase` 🙏

Reply:
Oh, yeah, good catch, @hoangnguyen92dn 👍 Thanks for reminding me 🙏

Reply:
Please kindly double-check by adding the mock response for `mockGetCoinPricesUseCase` 🙏

Reply:
Resolved 🚀 @hoangnguyen92dn Please check in this [commit](https://github.com/nimblehq/jetpack-compose-crypto/pull/96/commits/70f28eab01dd44ce45eab9c5db8b991935ec9c0c)

Reply:
@Wadeewee I just realized that the `returns mockk()` is applied for `Hierarchical mocking`, seems we should returns `flowOf(emptyList())` instead.

https://mockk.io/

![image](https://user-images.githubusercontent.com/6950766/220268542-91765c20-1777-487b-b8b6-38b06df0c487.png)

Reply:
Fixed @hoangnguyen92dn 🙏